### ISSUE TITLE
vagrant resume runs provisioners

### ISSUE THREAD

**Issue by hgraca** [mitchellh/vagrant#6787]
Hi,

Im experiencing what seems to be a bug:

I run ```vagrant up``` and it boots and provisions the box properly, using my customized box which already has most of my required packages installed.

I run ```vagrant suspend``` and it suspends the box properly.

I run ```vagrant resume``` and it resumes the box BUT it also starts running the provisioners.

Is this a bug or am I doing something wrong?
Any hint as to why this might be happening?

Using:
Vagrant 1.8.1
vbox 5.0.12r104815

This is my vagrantfile:

```rb
# -*- mode: ruby -*-
# vi: set ft=ruby :

required_plugins = %w(vagrant-bindfs vagrant-vbguest vagrant-triggers vagrant-s3auth)

plugins_to_install = required_plugins.select { |plugin| not Vagrant.has_plugin? plugin }
if not plugins_to_install.empty?
  puts "Installing plugins: #{plugins_to_install.join(' ')}"
  if system "vagrant plugin install #{plugins_to_install.join(' ')}"
    exec "vagrant #{ARGV.join(' ')}"
  else
    abort "Installation of one or more plugins has failed. Aborting."
  end
end

Vagrant.configure(2) do |config|
  config.vm.box = "<my_vendor>/base-trusty64"
  config.vm.box_check_update = false
  config.vm.box_url = "https://s3.amazonaws.com/<my-boxes>/base-trusty64.json"

  config.vm.network "private_network", ip: "172.16.0.100"

  if Vagrant.has_plugin?('vagrant-bindfs')
     config.vm.synced_folder "./www", "/mnt/vagrant", type: 'nfs'
     config.bindfs.bind_folder "/mnt/vagrant", "/var/www/<my_vendor>", owner: "<my_vendor>", group: "<my_vendor>", perms: "u=rwx:g=rwx:o=rwx", o: "nonempty"
  else
     exit;
  end

  config.vm.provider "virtualbox" do |vb|
    require 'time'
    vb.memory = "2048"
    vb.cpus = 4
    vb.name = "<my_vendor>.webshopapp.dev___" + Time.now.utc.iso8601
  end

  config.vm.provision "Fix for 'stdin: is not a tty'", type: "shell" do |s|
    s.inline = "sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n/' /root/.profile"
  end

  config.vm.provision "Provisioning", type: "shell"  do |s|
    s.path = 'var/setup/execute-files.sh'
  end

  config.trigger.after [:up,:reload,:resume] do
    run_remote  "/bin/bash /vagrant/var/sh/startup/restart-supervisor.sh"
  end

  config.trigger.before [:up,:reload] do |s|
    system('vagrant box outdated')
  end
end
```

**Comment by sbocconi** [comment]
I have the same issue, with the same version for Vagrant and vbox, and a Vagrant file generated with puphpet. What is worst is that some changes on disk are not persisted (like changing .bashrc), possibly due to the reprovision.

**Comment by westse** [comment]
I'm seeing this as well with v1.8.1

**Comment by brasileric** [comment]
Same problem with Vagrant 1.8.1:
vagrant resume -> runs provision

**Comment by jtreminio** [comment]
Doesn't fix the core issue, but after `vagrant suspend` I always run `vagrant up` to bring the VM back up, and provisioner is not run.

**Comment by jwarkentin** [comment]
This has happened to me on two different VMs now when resuming with 1.8.1.

**Comment by thiagotalma** [comment]
Here it is also going the same problem.
```
- Windows 10
- Virtualbox 5.0.14
- Vagrant 1.8.1
(Always running vagrant as administrator)
```

**Comment by yoke88** [comment]
the same problem .

1. vagrant suspend then vagrant resume ==> run provision
2. vagrant snapshot save init then vagrant snapshot restore init ==> run provision

I have to vagrant halt first and then go to vritualbox , restore the snapshot , then vagrant up.

**Comment by Tobion** [comment]
Confirm the problem vagrant resume runs provision.

**Comment by mauricios** [comment]
Same problem here, this is the workaround to resume and skip provision:

`vagrant up --no-provision`

**Comment by stefanhorning** [comment]
Yes, happens since I upgraded to Vagrant v1.8.1 today (provisioner never ran on resume with 1.7).

**Comment by davidsalazar** [comment]
![facepalm](https://cloud.githubusercontent.com/assets/14309/13885580/25f7beb2-ed01-11e5-8243-22184704ee7f.gif)

**Comment by cheeryfella** [comment]
Any progress on this since we generated the inifite loop with #7133 ?

**Comment by masi** [comment]
Same here. A coworker was wonderung why I complained about resume running a provisioner. It turned out he was running a 1.7.x. We agreed that he shouldn't upgrade until this issue is fixed.

**Comment by dvlancer** [comment]
is there an ETA on when this will be solved?

**Comment by thiagotalma** [comment]
> is there an ETA on when this will be solved?

Apparently the Vagrant was abandoned by @hashicorp :cry:

**Comment by sethvargo** [comment]
Vagrant has not been abandoned. We are a very small team and have not had the opportunity to test these new changes yet. Sorry!

**Comment by leoquijano** [comment]
Thanks guys for all your hard work. Vagrant is a hugely useful tool. @sethvargo

**Comment by lassehp** [comment]
Oh my! It seems I have been bitten by this bug, and damn it's a pain in the...

The documentation specifically says that resume does NOT run provisioners.

From https://www.vagrantup.com/docs/cli/resume.html :
> Command: vagrant resume [name|id]
> 
> This resumes a Vagrant managed machine that was previously suspended, perhaps with the suspend command.
> 
> The configured provisioners will not run again, by default. You can force the provisioners to re-run by specifying the --provision flag.

Also vagrant up was changed at some point (when I began using vagrant a couple years ago to require --run-provisioners to run provisioners. Before that, vagrant resume had to be used to resume a suspended vm without provisioning.)

As this is a destructive bug (I just lost a fine working development environment for a project), I hope this bug will get some attention, I am using 1.8.1, but as this issue remains open, I suspect it remains a problem. Or at LEAST edit the docs so the current destructiveness of resume is noted.

/Lasse, still feeling kicked in the guts and ass simultaneously
